be/src/storage/lake/transactions.cpp

```diff
@@ -53,9 +71,43 @@ static void clear_remote_snapshot_async(TabletManager* tablet_mgr, int64_t table
     files_to_delete->emplace_back(std::move(slog_path));
 }
 
+int64_t cal_new_base_version(int64_t tablet_id, TabletManager* tablet_mgr, int64_t base_version, int64_t new_version) {
+    int64_t version = base_version;
+    auto metadata = tablet_mgr->get_latest_cached_tablet_metadata(tablet_id);
+    if (metadata != nullptr && metadata->version() <= new_version) {
+        version = std::max(version, metadata->version());
+    }
+
+    auto index_version = tablet_mgr->update_mgr()->get_primary_index_data_version(tablet_id);
+    if (index_version > new_version) {
+        tablet_mgr->update_mgr()->try_remove_primary_index_cache(tablet_id);
```

Comment:
If `try_remove_primary_index_cache` fail because `ref` no equal 1, does it matters?

Reply:
And why no check whether the index version exists in remote storage. If not, clear and rebuild the index